public/model/epi-data-store.js

```diff
@@ -0,0 +1,143 @@
+    // like this:
+    // {start_time: <Date>, end_time: <Date>, region_case_data: <region case data>}
+    //
+    // The array is sorted by `start_time`. The times are all at midnight UTC.
+    // The time interval is half-open: [start_time, end_time). So, with
+    // start_time = 2016-01-01 and end_time = 2016-01-08, the interval contains
+    // data for 2016-01-01 up to and including 2016-01-07.
+    //
+    // Region case data is a mapping from region code -> condition ->
+    // culmulative case count. For example:
+    // {br1: {dengue: 120, malaria: 110, zika: 100},
+    //  br2: {dengue: 220, malaria: 210},
+    //  br3: {malaria: 310, zika: 300}}
+    this.data_by_date_interval_and_region = [];
+    this.most_recent_start_time = null;
+
+    this.initial_load_promise = Q.delay(10).then((function() {
+      // 4589 is Iguape, 4611 is Itanhaém, and 4877 is São Paulo.
+      this.data_by_date_interval_and_region = [
+        // Feb 1 - Feb 8: Missing malaria data for 4589.
+        {start_time: utc_date(2016, 2, 1), end_time: utc_date(2016, 2, 8),
+         region_case_data: {4589: {fake_dengue: 111},
+                            4611: {fake_dengue: 121, fake_zika: 122},
+                            4877: {fake_dengue: 1131, fake_zika: 1132}}
+        },
+        // Feb 8 - Feb 15: Missing dengue data for 4611. Also, 4877 has chikungunya data.
+        {start_time: utc_date(2016, 2, 8), end_time: utc_date(2016, 2, 15),
+         region_case_data: {4589: {fake_dengue: 211, fake_zika: 212},
+                            4611: {fake_zika: 222},
+                            4877: {fake_dengue: 1231, fake_zika: 1232, fake_chikungunya: 1233}}
+        },
+        // Feb 15 - Feb 29: Only has data for 4611. 2 weeks.
+        {start_time: utc_date(2016, 2, 15), end_time: utc_date(2016, 2, 29),
+         region_case_data: {4611: {fake_dengue: 321, fake_zika: 322}}
+        },
+        // Feb 15 - Mar 8: Only has data fro 4589 and 4877. Overlaps with previous time span,
+        // but longer.
+        {start_time: utc_date(2016, 2, 15), end_time: utc_date(2016, 3, 8),
+         region_case_data: {4589: {fake_dengue: 311, fake_zika: 312},
+                            4877: {fake_dengue: 1331, fake_zika: 1332}}
+        }
+      ];
+      this.most_recent_start_time = _.last(this.data_by_date_interval_and_region).start_time;
+    }).bind(this));
+  },
+
+  // Return a recent epi data record. If there are multiple records that include
+  // the given `date`, returns the one that has data on the most regions. If
+  // there are no records that include `date`, returns the most recent record
+  // that occurred before the date.
+  get_best_recent_epi_data: function(date) {
+    // TODO(jetpack): binary search instead of linear scan.
+    var matching_records = _.filter(this.data_by_date_interval_and_region, function(record) {
+      return record.start_time <= date && date < record.end_time;
```

Comment:
I imagine `<` does really funky things in JS if the types of these are different. Maybe add some type checks before?